### ISSUE TITLE
Opening files from explorer with keypress

### ISSUE THREAD

**Issue by NiloCK** [zed-industries/zed#17646]
### Check for existing issues

- [X] Completed

### Describe the feature

Maybe a silly one and probably a skill or configuration issue, but:

- `ctrl-shift-e`: open file explorer
- `up`, `down` arrows: navigate to desired file
- `enter`: edit filename?

Expected bahaviour is to open the selected file for editing.

Ubuntu, VSCode keybindings, Zed 0.151.2.

### If applicable, add mockups / screenshots to help present your vision of the feature

[Screencast from 2024-09-10 10:42:03 AM.webm](https://github.com/user-attachments/assets/07936cb4-fa9d-4915-8456-80ffb2a7ae45)

**Comment by CharlesChen0823** [comment]
I think this is work as expect.

**Comment by NiloCK** [comment]
Wow. Thanks much @jansol, @CharlesChen0823. I've never really used Finder or macOS. I'm now curious whether VSCode on macOS opens a file or edits filenames on `enter`.

If it opens the file, there may be a case for having Zed do the same when configured with VSCode keymap. Will leave this issue open in case it wants to be repurposed toward that. Please close otherwise.

In my fingers, `F2` is for rename (and works in Zed).

"Cannot open file" was approaching "cannot exit vim" levels of helplessness for me :)

**Comment by JosephTLyons** [comment]
Hey @NiloCK, I'm betting you are using the default keymap, which is modeled after VS Code's default key bindings:

<img width="550" alt="SCR-20240910-lkqj" src="https://github.com/user-attachments/assets/ed51d1e7-d03c-41cc-9a4d-2a023f1de637">

VS Code uses `spacebar` to open a file when it is highlighted, so that's what we went with as well.

You can swap the actions around by adding the following to your `keymap.json`:

```json
{
  "context": "ProjectPanel && not_editing",
  "bindings": {
    "enter": "project_panel::Open"
  }
},
{
  "context": "ProjectPanel",
  "bindings": {
    "space": "project_panel::Rename"
  }
}
```

**Comment by JosephTLyons** [comment]
I guess I missed the fact that this is exclusively about Zed on Linux - whatever VS Code does on Linux, we should be following in the default keycap, and if we aren't, we should change it.

**Comment by NiloCK** [comment]
> I guess I missed the fact that this is exclusively about Zed on Linux - whatever VS Code does on Linux, we should be following in the default keycap, and if we aren't, we should change it.

Yes - this is it. Unless my muscle memory and general cognition have gone entirely out the window, VSCode on Linux opens files from the explorer here with `enter`.

For completeness - I just checked - VSCode also opens a file with `space`.

**Comment by Haardt** [comment]
Maybe you like also:
 ```
 {
    "context": "ProjectPanel && not_editing",
    "bindings": {
      "enter": "project_panel::Open",
      "escape": ["workspace::ActivatePane", 0]
    }
```

**Comment by jakubsuchybio** [comment]
> Maybe you like also:
> 
> ```
> {
>    "context": "ProjectPanel && not_editing",
>    "bindings": {
>      "enter": "project_panel::Open",
>      "escape": ["workspace::ActivatePane", 0]
>    }
> ```

Yeah I was just figuring out how to do multi-command shortcut for these 2 actions, but having it on esc is good enough. Sad that multi-command actions are not supported. I have read only about workspace::SendKeystrokes, but with that I couldn't figure out how to do those 2 actions on "enter"

**Comment by notpeter** [comment]
Thanks all, I'm going to go close as unactionable. You can remap alternate keys if you prefer.

Thanks for reporting.
